firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -508,12 +509,12 @@ private static DocumentReference convertDocumentReference(Object o, ErrorPath pa
     }
   }
 
-  private static RuntimeException serializeError(ErrorPath path, String reason) {
+  private static IllegalArgumentException serializeError(ErrorPath path, String reason) {
     reason = "Could not serialize object. " + reason;
     if (path.getLength() > 0) {
       reason = reason + " (found in field '" + path.toString() + "')";
     }
-    return new RuntimeException(reason);
+    return new IllegalArgumentException(reason);
```

Comment:
Some errors (in particular trying to use an Array instead of a List) are now getting caught in the POJO mapper instead of our UserDataConverter.  To keep the exception type consistent, I CustomClassMapper to throw IllegalArgumentExceptions (which is more correct anyway).